lucene/queryparser/build.gradle

```diff
@@ -7,3 +7,305 @@ dependencies {
 
   testImplementation project(':lucene:test-framework')
 }
+
+configure(":lucene:queryparser") {
+  configurations {
+    javaCCDeps
+
+    dependencies {
+      javaCCDeps "net.java.dev.javacc:javacc:5.0"
+    }
+  }
+}
+def conf = [
+    "queryParser": ["inputFile": file('src/java/org/apache/lucene/queryparser/classic/QueryParser.jj'), "outputDir": file('src/java/org/apache/lucene/queryparser/classic')],
+    "surround"   : ["inputFile": file('src/java/org/apache/lucene/queryparser/surround/parser/QueryParser.jj'), "outputDir": file('src/java/org/apache/lucene/queryparser/surround/parser/')],
+    "flexible"   : ["inputFile": file('src/java/org/apache/lucene/queryparser/flexible/standard/parser/StandardSyntaxParser.jj'), "outputDir": file('src/java/org/apache/lucene/queryparser/flexible/standard/parser')]
+]
+
+String javaCCClasspath
+String javaCCHome
+
+task installJavacc {
+  doLast {
+    javaCCClasspath = project.project(":lucene:queryparser").configurations.javaCCDeps.asPath
+    javaCCHome = javaCCClasspath.substring(0, javaCCClasspath.lastIndexOf("/"))
+
+    String hacky = javaCCHome + "/" + "javacc.jar"
+    File hackyFile = new File(hacky)
+    println hackyFile
+    println hackyFile.exists()
+    if (hackyFile.exists() == false) {
+      hackyFile.bytes = new File(javaCCClasspath).bytes
+      hackyFile.setExecutable(true)
+    }
+  }
+}
+String lineSeparator = System.lineSeparator()
+
+task cleanOldGeneratedFiles {
+  doLast {
+    println "Removing old generated files"
+    conf.each { key, val ->
+      val["outputDir"].eachFileMatch ~/.*\.java/, { f ->
+        if (f.text =~ /Generated.*By.*JavaCC/) {
+          f.delete()
+        }
+      }
+    }
+  }
+}
+
+task runJavaccQueryParser(dependsOn: installJavacc) {
```

Comment:
I think because there is so much commonality between all of the javacc tasks, I would create a `task JavaCC` class and put all the clean up logic in one place rather than repeating it.

Reply:
Gah, it's worse than that. The common bits _are_ in one place, see cleanUpAfter. I missed taking it out of a place or two. Fixing it up.

So there's only one small bit that's repeated now, I don't think it's worth a new class now.

And many thanks for your review! I'll get the hang of this yet.